web-locks/non-fully-active.tentative.https.html

```diff
@@ -0,0 +1,73 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>Web Locks API: Non-fully-active documents</title>
+<link rel=help href="https://wicg.github.io/web-locks/">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/helpers.js"></script>
+
+<div></div>
+
+<script>
+  function createNonFullyActiveIframe(src) {
+    const iframe = document.createElement("iframe");
+    if (src) {
+      iframe.src = src;
+    }
+    document.body.appendChild(iframe);
+    const { navigator, DOMException, postMessage } = iframe.contentWindow;
+    iframe.remove();
+    return { iframe, navigator, DOMException, postMessage };
+  }
+
+  promise_test(async t => {
+    const { navigator, DOMException } = createNonFullyActiveIframe();
+    const p = navigator.locks.request("foo", () => { });
+    await promise_rejects_dom(t, "InvalidStateError", DOMException, p, "Request should explicitly fail");
+  }, "request() on non-fully-active document must fail");
+
+  promise_test(async t => {
+    const { navigator, DOMException } = createNonFullyActiveIframe();
+    const p = navigator.locks.query();
+    await promise_rejects_dom(t, "InvalidStateError", DOMException, p, "Query should explicitly fail");
+  }, "query() on a non-fully-active document must fail");
+
+  promise_test(async t => {
+    const { navigator, DOMException, postMessage } = createNonFullyActiveIframe("resource/iframe.html");
+
+    const p = navigator.locks.request("-", () => { });
+    await promise_rejects_dom(t, "InvalidStateError", DOMException, p, "Request should explicitly fail");
+  }, "request()'s fully-active check happens earlier than name validation");
+
+  promise_test(async t => {
+    const { iframe, navigator, DOMException } = createNonFullyActiveIframe();
+    document.body.append(iframe);
+    t.add_cleanup(() => iframe.remove());
+
+    assert_not_equals(navigator, iframe.contentWindow.navigator, "Navigator object changes");
+    assert_not_equals(navigator.locks, iframe.contentWindow.navigator.locks, "LockManager object changes");
+
+    const p = navigator.locks.request("foo", () => { });
+    await promise_rejects_dom(t, "InvalidStateError", DOMException, p, "Request on the previous LockManager still must fail");
+  }, "Reactivated iframe must not reuse the previous LockManager");
+
+  promise_test(async t => {
+    const iframe = document.createElement("iframe");
+    document.body.appendChild(iframe);
+    const worker = new iframe.contentWindow.Worker("resources/worker.js");
+
+    const res = uniqueName(t);
```

Comment:
Nit, I have no idea what 'res' means.

Reply:
`res` is used by quite a lot of web locks tests and means "resource name", but I guess `name` is fine.